llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2893,6 +2904,18 @@ bool isKnownNonZero(const Value *V, const APInt &DemandedElts, unsigned Depth,
     }
   }
 
+  std::optional<ConstantRange> Range;
+  if (const CallBase *CB = dyn_cast<CallBase>(V))
+    Range = CB->getRange();
```

Comment:
Could you please rebase over https://github.com/llvm/llvm-project/commit/2cc75aed0925f3c7e8e50b2f4bdc94daac4cb08d and move this part into the CallBase handling code in isKnownNonZeroFromOperator()?